robot_ws/src/voice_interaction_robot/test/integration_test

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python
+"""
+ Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ software and associated documentation files (the "Software"), to deal in the Software
+ without restriction, including without limitation the rights to use, copy, modify,
+ merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+ permit persons to whom the Software is furnished to do so.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+"""
+
+
+"""
+    This script runs a series of integration tests for the voice_interaction node. 
+    It can be launched with the integration_test launch file and it will output
+    how many tests are to be run and the results for each test. 
+"""
+
+import rospy
+import rosnode
+import rostopic
+import rosservice
+import time
+from std_msgs.msg import String
+from geometry_msgs.msg import Twist, Vector3
+
+
+class VoiceInteractionIntegrationTest:
+    wake_words = ("jarvis", "turtlebot")
+    last_cmd_vel = None
+    vinode_start_timeout = 10
+    
+    def __init__(self, test_sleep_time=1):
+        rospy.init_node("integration_test", disable_signals=True)
+        self.text_input_publisher = rospy.Publisher("/text_input", String, queue_size=5)
+        self.wake_publisher = rospy.Publisher("/wake_word", String, queue_size=5)
+        rospy.Subscriber("/cmd_vel", Twist, self.save_cmd_vel)
+        self.test_sleep_time = test_sleep_time
+        
+    def run_tests(self):
+        self.wait_for_voice_interaction_nodes()
+        self.wait_for_voice_interaction_services()
+        self.wait_for_voice_interaction_node_to_subscribe_to_topic()
+        self.run_text_input_tests()
+        
+    def wait_for_voice_interaction_nodes(self):
+        required_nodes = set([
+            '/lex_node', 
+            '/voice_interaction', 
+            '/voice_command_translator'
+        ])
+        while not required_nodes.issubset(rosnode.get_node_names()):
+            time.sleep(0.1)
```

Comment:
Just tried this and the unfortunate downside is that if there is no simulation world running the program hangs on `rospy.time()`, so it seems like the test is broken. I don't think it's necessary to use rospy.time for this.

Reply:
I think this is because you have use_sim_time=true in the integraiton_test launch file. That causes rospy.time to listen to the /clock topic for time instead of using the system clock. Without a simulation world nothing is publishing to the /clock topic so it makes sense that it'd hang. 

Since your tests aren't interacting with the simulation world I think you're right that it doesn't really matter in this case.